### ISSUE TITLE
Plan to improve "memetic" evolution

### ISSUE THREAD

**Issue by nleck** [stSoftwareAU/NEAT-TS#1321]
Please study our implementation of "memetic" evolution and create GH issues and sub issues to improve ( if possible)

The "memetic" evolution is working very well and for some reason seemingly better than back propagation, which is odd in itself. 

Can we made memetic evolution better? 

Each issue should make the standard reminders including TDD and DRY. Performance (speed) should include benchmarks as evidence. Algorithmic improvements should include unit tests

**Comment by stservice** [comment]
## Memetic Evolution Analysis Complete

I've studied the memetic evolution implementation and created the following sub-issues for potential improvements:

### Fine-Tuning Enhancements
1. #1323 - **Adaptive fine-tuning population size** - Dynamically adjust based on success rate
2. #1330 - **Configurable quantum step size** - Adaptive step sizing based on training progress
3. #1331 - **Bias-weight coordination** - Coordinated quantum adjustments for efficiency

### Memetic Data Management
4. #1324 - **Multi-generational ancestral learning** - Track multiple generations of weight history
5. #1326 - **Partial memetic inheritance** - Preserve memetic data across structural mutations

### Algorithm Improvements
6. #1325 - **Gradient-informed quantum adjustment** - Use backprop gradient information
7. #1327 - **Intelligent retry selection** - Learn from historical retry success/failure
8. #1328 - **Species-aware fine-tuning comparison** - Better comparison target selection
9. #1329 - **Coupling fine-tuning with discovery** - Integrate memetic and discovery processes

### Diagnostics
10. #1332 - **Memetic statistics and diagnostics** - Comprehensive tracking for analysis

---

### Architecture Overview

The current memetic evolution system works well due to:

1. **Elegant quantum discretisation** - MIN_STEP prevents floating-point creep
2. **Elastic backpropagation** - Minimum-change heuristic with safe-zone awareness
3. **Multi-source fine-tuning** - Comparisons with previous fittest, species members, and restored sources
4. **Non-destructive exploration** - Generates variants without modifying originals
5. **Retry/backtrack mechanism** - Can recover from wrong directions

The proposed improvements aim to enhance these strengths while addressing identified opportunities:
- Better utilisation of historical learning
- More intelligent selection of comparison targets
- Adaptive parameters based on training progress
- Tighter integration with structural discovery

Each issue includes TDD, DRY, and performance benchmark requirements per project guidelines.

**Comment by stservice** [comment]
## Automated Processing Failed (First Attempt)

The automated worker was unable to complete this issue on the first attempt.

### Failure Details
Git push failed while pushing branch `issue-1321-plan-to-improve-memetic-evolution` to origin.

This typically happens due to:
- Authentication/permission issues
- Protected branch rules
- Network connectivity problems
- Missing `workflow` scope for OAuth tokens (when modifying GitHub Actions files)

<details>
<summary>Git push error output (click to expand)</summary>

```
To https://github.com/stSoftwareAU/NEAT-AI.git
 ! [rejected]          issue-1321-plan-to-improve-memetic-evolution -> issue-1321-plan-to-improve-memetic-evolution (fetch first)
error: failed to push some refs to 'https://github.com/stSoftwareAU/NEAT-AI.git'
hint: Updates were rejected because the remote contains work that you do not
hint: have locally. This is usually caused by another repository pushing to
hint: the same ref. If you want to integrate the remote changes, use
hint: 'git pull' before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```
</details>

### What happens next?
- This issue has been marked with the `failed-once` label
- The worker will **automatically retry** this issue on the next scan
- If it fails again, the issue will be marked as `failed` and require manual intervention

### To prevent automatic retry
If you want to work on this manually, remove the `failed-once` label and assign yourself.

**Comment by stservice** [comment]
## Automated Processing Failed (Second Attempt - Permanently Failed)

The automated worker has failed to complete this issue twice and will not retry automatically.

### Failure Details
Git push failed while pushing branch `issue-1321-plan-to-improve-memetic-evolution` to origin.

This typically happens due to:
- Authentication/permission issues
- Protected branch rules
- Network connectivity problems
- Missing `workflow` scope for OAuth tokens (when modifying GitHub Actions files)

<details>
<summary>Git push error output (click to expand)</summary>

```
To https://github.com/stSoftwareAU/NEAT-AI.git
 ! [rejected]          issue-1321-plan-to-improve-memetic-evolution -> issue-1321-plan-to-improve-memetic-evolution (fetch first)
error: failed to push some refs to 'https://github.com/stSoftwareAU/NEAT-AI.git'
hint: Updates were rejected because the remote contains work that you do not
hint: have locally. This is usually caused by another repository pushing to
hint: the same ref. If you want to integrate the remote changes, use
hint: 'git pull' before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```
</details>

### What happened?
- This issue failed on the first attempt and was marked with `failed-once`
- The worker retried and failed again on the second attempt
- The issue has been marked with the `failed` label to prevent repeated attempts
- Manual intervention is required

### Why this might be happening
- The issue may be too complex or large for automated processing
- The issue description may need more detail or clarification
- There may be dependencies or permissions that Claude doesn't have

### To retry this issue
1. Investigate and address the underlying problem (check the failure details above)
2. Consider simplifying or breaking down the issue into smaller tasks
3. Remove the `failed` label from this issue
4. The worker will pick it up again on the next scan
